File: Frontend/Logic/B20/Infrastructure/HttpClientModule/Context/Impl.cs
using B20.Architecture.ContextModule.Api;
using B20.Infrastructure.HttpClient.Integrations;
using HttpClientModule.Api;
using HttpClientModule.Impl;

namespace B20.Infrastructure.HttpClientModule.Context
{
    public class HttpClientModuleImpl: ContextModule
    {
        public void Apply(ContextBuilder builder)
        {
            builder.SetImpl<HttpClientFactory, HttpClientFactoryLogic>();
        }
    }
    
    public class DotNetHttpClientModuleImpl: ContextModule
    {
        public void Apply(ContextBuilder builder)
        {
            builder
                .WithModule(new HttpClientModuleImpl())
                .SetImpl<HttpRequester, DotNetHttpRequester>();
        }
    }
}

File: Backend/lib/src/main/kotlin/com/github/bratek20/ostium/kajiugame/context/Impl.kt
package com.github.bratek20.ostium.kajiugame.context

import com.github.bratek20.architecture.context.api.ContextBuilder
import com.github.bratek20.architecture.context.api.ContextModule

import com.github.bratek20.ostium.kajiugame.api.*
import com.github.bratek20.ostium.kajiugame.impl.*

class KajiuGameImpl: ContextModule {
    override fun apply(builder: ContextBuilder) {
        builder
            .setImpl(GameApi::class.java, GameApiLogic::class.java)
    }
}